### PR TITLE
JP Pro Dashboard: Add the license lightbox in the Issue License screen.

### PR DESCRIPTION
**Pull request by jkguidaven**
> This PR is part of a 2-part PR that aims to bring the Product Lightbox to the Jetpack agency Pro Dashboard - Issue License screen. The first PR can be located here: #76760

This PR updates the Issue License screen to display the product information when pressing the more about links in the product or bundle card. 

Screen Shot 2023-05-29 at 3 56 18 PM

Screen Shot 2023-05-29 at 3 55 26 PM

Additionally, it was planned to reuse the existing product lightbox component on the pricing page. However, due to the component's complexity and how tightly coupled it is to the checkout flow, I decided it would be better to create a separate lightbox to reduce the risk of breaking something on the pricing page if we try to tweak it for the agency's needs.

In this PR, I created a dumb version of the product lightbox that is much more reusable than the previous one. I also moved the component to the shared component folder, given that this will be used on multiple codebases.

I also plan to update the product lightbox on the pricing page to use the new lightbox on a separate PR to avoid bloating the size of this already huge PR.

Related to 1202619025189113-as-1203886504803361

## Proposed Changes

*  Add a dumb version that can be reused for the pricing page and the agency dashboard. The new lightbox component should also allow flexibility on what content to render. 
* Add a `JetpackProductInfo` component that will render product info. The idea is, later, when we decide on new Product information specific to the agency dashboard, we can now easily swap it with a different component. I plan to reuse this component for the product lightbox on the pricing page on a separate PR.
* Implement the `LicenseLightbox` component that utilizes the new reusable components.

## Testing Instructions
* Make sure you have an agency account before testing this.
* Use the Jetpack cloud live link below and append `/partner-portal/issue-license.`
* In the issue license screen. Click any 'more about' link and confirm the lightbox shows up.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1203886504803361

### CODE REVIEW

**Review by ilonagl — APPROVED** [review]
Looks very good! Amazing job.

**Review by yashwin — CHANGES_REQUESTED** [review]
As discussed, let's reuse the code which are common in this PR and move those into a common folder in an upcoming PR 🙂

**Comment by jkguidaven** [comment]
> As discussed, let's reuse the code which are common in this PR and move those into a common folder in an upcoming PR 🙂

Thanks, @yashwin, for the suggestion. I made the necessary changes to reuse existing files instead of making one. This should reduce the PR size 👍

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/hooks/use-license-lightbox-data.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		const title = getProductTitle( product.name );
+
+		if ( productInfo ) {
+			return {
+				title,
+				product: productInfo,
+			} as LicenseLightboxData;
+		}
+
+		return { title } as LicenseLightboxData;
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/hooks/use-license-lightbox-data.ts]: I don't think `as LicenseLightboxData` is required in both returns. We can remove that and set 

```
type LicenseLightboxData = {
	title: string;
	product:? SelectorProduct;
};
```

WDYT?

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/style.scss
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox/style.scss:1]: Could we please
- use variables instead of using color codes(ex: `#fff`) directly
- use RTL specific CSS attributes instead of using attributes like `margin-bottom`

File: client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx [hunk 2]
```diff
@@ -1,58 +1,26 @@
...
-				<img className="license-lightbox-link__icon" src={ ModalLinkIcon } alt="" />
-			</Button>
+		<Button className="license-lightbox-link" plain onClick={ onClick }>
+			{ translate( 'More about {{productName/}}', {
+				components: { productName: <>{ productName }</> },
+			} ) }
 
-			{ showLightbox && <LicenseLightbox onClose={ onHideLightbox } /> }
-		</>
+			<img className="license-lightbox-link-icon" src={ ModalLinkIcon } alt="" />
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx:22]: Let's not keep the `alt` empty.

File: client/jetpack-cloud/sections/partner-portal/license-bundle-card/index.tsx [hunk 3]
```diff
@@ -17,45 +21,77 @@ interface Props {
...
+	}, [ onSelectProduct, product ] );
 
 	const { description: productDescription } = useProductDescription( product.slug );
 
+	const onShowLightbox = useCallback(
+		( e: React.MouseEvent< HTMLElement > ) => {
+			e.stopPropagation();
+
+			dispatch(
+				recordTracksEvent( 'calypso_partner_portal_issue_license_product_view', {
```
Comment (yashwin) [inline client/jetpack-cloud/sections/partner-portal/license-bundle-card/index.tsx]: Can we please keep the event name something like `calypso_partner_portal_issue_license_bundle_product_view` so we can differentiate between product and bundle?

File: client/components/jetpack/jetpack-product-info/faq-list.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	const onToggle = useCallback(
+		( faqArgs: { id: string; isExpanded: boolean } ) => {
+			const { id, isExpanded } = faqArgs;
+			const tracksArgs = {
+				faq_id: id,
+			};
+
+			dispatch(
+				recordTracksEvent(
+					isExpanded ? 'calypso_plans_faq_open' : 'calypso_plans_faq_closed',
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/faq-list.tsx]: I think we can keep the event names as `calypso_partner_portal_product_details_faq_open` & `calypso_partner_portal_product_details_faq_closed`. WDYT?

File: client/components/jetpack/jetpack-product-info/index.tsx [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	const translate = useTranslate();
+	const icon = getProductIcon( { productSlug } );
+
+	const descriptionMap = useIncludedProductDescriptionMap( product.productSlug );
+
+	return (
+		<div className="jetpack-product-info">
+			<div className="jetpack-product-info__header">
+				<div className="jetpack-product-info__product-icon">
+					<img alt="" src={ icon } />
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/index.tsx:37]: Let's not keep the `alt` empty

File: client/components/jetpack/jetpack-product-info/product-list.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+const JetpackProductInfoProductListItem: React.FC< JetpackProductInfoProductListItemProps > = ( {
+	productSlug,
+	descriptionMap,
+} ) => {
+	const product = slugToSelectorProduct( productSlug );
+	return (
+		<div className="jetpack-product-info__product-list-item">
+			<div className="jetpack-product-info__product-list-item-icon">
+				<img alt="" src={ getProductIcon( { productSlug } ) } />
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/product-list.tsx]: Let's not keep the `alt` empty

File: client/components/jetpack/jetpack-product-info/recommendation-tags.tsx [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+	tags: tags,
+} ) => {
+	const translate = useTranslate();
+
+	return (
+		<div className="jetpack-product-info__tags">
+			<span className="jetpack-product-info__tags-label">{ translate( 'Great for:' ) }</span>
+
+			<ul className="jetpack-product-info__tags-list">
+				{ tags.map( ( tag ) => (
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/recommendation-tags.tsx]: How about we destructure the tag?

```
{ tags.map( ( { tag, label } ) => (
	<li className="jetpack-product-info__tags-list-item" key={ tag }>
		<span aria-hidden="true">{ Tags[ tag ] }</span>
		<p>{ label }</p>
	</li>
) ) }
```

File: client/components/jetpack/jetpack-product-info/style.scss
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/style.scss:1]: Could we please
- use variables instead of using color codes(ex: `#fff`) directly
- use RTL specific CSS attributes instead of using attributes like `margin-bottom`

File: client/components/jetpack/jetpack-lightbox/style.scss
Comment (yashwin) [inline client/components/jetpack/jetpack-lightbox/style.scss:1]: Could we please
- use variables instead of using color codes(ex: `#fff`) directly
- use RTL specific CSS attributes instead of using attributes like `margin-bottom`

File: client/components/jetpack/jetpack-lightbox/index.tsx [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+} ) => {
+	const translate = useTranslate();
+
+	return (
+		<Modal
+			className={ classNames( 'jetpack-lightbox__modal', className ) }
+			overlayClassName="jetpack-lightbox__modal-overlay"
+			isOpen={ isOpen }
+			onRequestClose={ onClose }
+			htmlOpenClassName="ReactModal__Html--open lightbox-mode"
```
Comment (yashwin) [inline client/components/jetpack/jetpack-lightbox/index.tsx]: Can we use the proper class name convention instead of `ReactModal__Html--open` here?

File: client/components/jetpack/jetpack-lightbox/index.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+					plain
+					onClick={ onClose }
+					aria-label={
+						translate( 'Close', {
+							comment:
+								'Text read by screen readers when the close button of the lightbox gets focus.',
+						} ) as string
+					}
+				>
+					{ CloseIcon }
```
Comment (yashwin) [inline client/components/jetpack/jetpack-lightbox/index.tsx]: I think we can use the existing icons.

```
import { Icon, close } from '@wordpress/icons';
...
<Icon size={ 24 } icon={ close } />
```

File: client/components/jetpack/jetpack-lightbox/index.tsx [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+const JetpackLightbox: FunctionComponent< JetpackLightboxProps > = ( {
+	children,
+	className,
+	isOpen,
+	onClose,
+} ) => {
+	const translate = useTranslate();
+
+	return (
+		<Modal
```
Comment (yashwin) [inline client/components/jetpack/jetpack-lightbox/index.tsx:51]: Modal is imported from `'react-modal'`. Is there a quick way to use it from the `'@wordpress/components'` so we don't have to specify the close icon explicitly? We could do this in another PR.

**Review by yashwin — CHANGES_REQUESTED** [review]
Good job on this, @jkguidaven! 

I have left some comments.

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/hooks/use-license-lightbox-data.ts (hunk 1 above)
Comment (jkguidaven) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/hooks/use-license-lightbox-data.ts]: Excellent suggestion, @yashwin 👍. This one has been applied.

File: client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx (hunk 2 above)
Comment (jkguidaven) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx:22]: Thanks for raising this, @yashwin; I learned from the previous accessibility task that we intend to set alt as empty for decorative icons. 

https://www.portent.com/blog/design-dev/decorative-images-how-to-identify-and-tag-them-for-accessibility.htm#:~:text=To%20do%20this%2C%20simply%20provide,out%20of%20the%20reading%20flow.

File: client/jetpack-cloud/sections/partner-portal/license-bundle-card/index.tsx (hunk 3 above)
Comment (jkguidaven) [reply client/jetpack-cloud/sections/partner-portal/license-bundle-card/index.tsx]: Good catch @yashwin. this one has been fixed. 👍

File: client/components/jetpack/jetpack-product-info/faq-list.tsx (hunk 4 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-product-info/faq-list.tsx]: Good point, since this component will eventually be used elsewhere besides the Partner portal, I renamed the event to`calypso_product_details_faq_open` and `calypso_product_details_faq_closed` instead. This is closer to a suggestion but is not specific to partner portal. wdyt?

File: client/components/jetpack/jetpack-lightbox/index.tsx (hunk 9 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-lightbox/index.tsx]: Excellent suggestion, @yashwin. I don't know why we previously used a custom icon instead of reusing the WordPress icon, but it might have something to do with the bit of variation on the icon. I recheck it, and I can see it almost looks the same.

File: client/components/jetpack/jetpack-lightbox/index.tsx (hunk 8 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-lightbox/index.tsx]: Absolutely, this has been fixed and renamed `jetpack-lightbox__html--is-open` which follows our Calypso convention.

File: client/components/jetpack/jetpack-product-info/recommendation-tags.tsx (hunk 7 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-product-info/recommendation-tags.tsx]: Nice suggestion. this has been applied 👍

File: client/components/jetpack/jetpack-lightbox/index.tsx (hunk 10 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-lightbox/index.tsx:51]: I have no context why the team previously decided to import the `react-modal` directly, but possibly it might have something to do with flexibility in customization. I agree this should be looked on a separate PR 👍

File: client/jetpack-cloud/sections/partner-portal/license-lightbox/style.scss
Comment (jkguidaven) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox/style.scss:1]: Thanks for raising this, @yashwin; I have updated the CSS to be RTL-friendly and use existing color variables. While I determined few of the colors does not have existing variable and are solely used for this component, I left it as is. Will try to figure out where we can best put this so it can be reused again later.

**Comment by jkguidaven** [comment]
@yashwin, thanks for the thorough review. I made the changes you suggested. Hopefully, I covered everything. 🙏

File: client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx (hunk 2 above)
Comment (yashwin) [reply client/jetpack-cloud/sections/partner-portal/license-lightbox-link/index.tsx:22]: Thanks for sharing this.

File: client/components/jetpack/jetpack-product-info/faq-list.tsx (hunk 4 above)
Comment (yashwin) [reply client/components/jetpack/jetpack-product-info/faq-list.tsx]: This is good 👍

**Review by yashwin — APPROVED** [review]
Thanks for making all the suggested changes, @jkguidaven. Good job on this! 

LGTM now.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7989035

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7989035).

**Comment by jkguidaven** [comment]
Hi @yashwin, can you review this PR again? There are a few changes in the pricing page's lightbox that I need to include in this PR. 

You don't need to look over the entire code again. Just the last commit should be good. 🙏

**Comment by jkguidaven** [comment]
Hi @Automattic/jetpack-martech , we are trying to bring the product lightbox into the Agency's license portal. We initially wanted to reuse the existing lightbox on the pricing page, but we ended up making a separate lightbox for the agency due to the risk of breaking something on the pricing page.

This PR is an attempt to unify the pricing page and Agency license portal to have a common product lightbox. However, for now, we are duplicating some of the codes (especially the styling) and moving them to the common package on this PR. A follow-up PR will focus on cleaning up the old lightbox and updating it so it can reuse the components being rebuilt here.

It would be nice if we could have someone from your team also have a look.

File: client/components/jetpack/jetpack-product-info/style.scss [hunk 11]
```diff
@@ -188,11 +202,38 @@
...
+.jetpack-product-info__product-list-item-callout {
+	padding: 0.25rem 0 0 0;
+	box-sizing: border-box;
+	color: var(--studio-gray-70);
+	font-size: 0.875rem;
+
+	@media ( min-width: $break-medium ) {
+		padding: 0.25rem;
+		min-width: 160px;
+		background-color: #eaefef;
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/style.scss:230]: Can we use variable here?

File: client/components/jetpack/jetpack-product-info/index.tsx [hunk 12]
```diff
@@ -42,10 +43,18 @@ const JetpackProductInfo: FunctionComponent< JetpackProductInfoProps > = ( {
...
-					descriptionMap={ descriptionMap }
-				/>
+				<>
+					<JetpackProductInfoProductList
+						products={ product.productsIncluded }
+						descriptionMap={ descriptionMap }
+					/>
+
+					{ alsoIncluded?.length && (
+						<JetpackProductInfoSection title={ translate( 'Also included:' ) }>
```
Comment (yashwin) [inline client/components/jetpack/jetpack-product-info/index.tsx:53]: I'm curious about this one.

Will `translate( 'Also included:' )` be translated considering the fact the translations are done for this PR?

**Review by yashwin — COMMENTED** [review]
LGTM and changes work as expected.

However, I have left a few comments. 

Can we please also include a screenshot of the bundle product details here?

File: client/components/jetpack/jetpack-product-info/index.tsx (hunk 12 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-product-info/index.tsx:53]: This one should already be translated. I just copied this string from the old lightbox. 👍

File: client/components/jetpack/jetpack-product-info/style.scss (hunk 11 above)
Comment (jkguidaven) [reply client/components/jetpack/jetpack-product-info/style.scss:230]: Good catch, @yashwin 👍 . Right now, this doesn't seem to have an existing variable. We will have to figure out where we can best put the variables so they can be reused, as there are a couple of colors we will also have to fix with similar situations. I am planning to fix it in the follow-up PR. For now, we can leave it as is.

**Comment by jkguidaven** [comment]
> Can we please also include a screenshot of the bundle product details here?

Thanks for the suggestion. I added it in the PR description. 👍

**Review by yashwin — APPROVED** [review]
Thanks for addressing the comments, @jkguidaven. LGTM now 🚀 🚢

**Review by robertsreberski — APPROVED** [review]
The PR tests well and the code looks great! I see how we can benefit from unifying this code as a standalone component in the future. How you made it modularized already makes it more maintainable for the future. Thanks for the effort!
